rdbi-recipes/src/main/java/com/lithium/dbi/rdbi/recipes/scheduler/MultiChannelScheduler.java

```diff
@@ -0,0 +1,217 @@
+package com.lithium.dbi.rdbi.recipes.scheduler;
+
+import com.lithium.dbi.rdbi.Handle;
+import com.lithium.dbi.rdbi.RDBI;
+import redis.clients.jedis.Tuple;
+
+import java.util.List;
+import java.util.Set;
+import java.util.function.LongSupplier;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+/**
+ * Similar to {@link StateDedupedJobScheduler}, except that it includes a separate "channel"
+ * dimension and attempts to maintain fairness among jobs in a particular by round-robining
+ * through all channels.
+ *
+ * Includes support for methods from {@link StateDedupedJobScheduler} and it's abstract hierarchy,
+ * however does not extend from that tree because some methods require the channel parameter
+ *
+ * it may prove useful to create a MultiChannelScheduler.ForChannel that extends from the {@link AbstractDedupJobScheduler} hierarchy
+ *
+ * // TODO This doesn't yet incorporate concepts from https://github.com/lithiumtech/rdbi/commit/6bbf2eeb49b87b71655f24fa9b797300b37b6797, that will be tackled separately
+ */
+public class MultiChannelScheduler {
+    private final RDBI rdbi;
+    private final String prefix;
+    private final LongSupplier clock;
+
+    public MultiChannelScheduler(RDBI rdbi, String redisPrefixKey, LongSupplier clock) {
+        this.rdbi = rdbi;
+        this.prefix = redisPrefixKey;
+        this.clock = clock;
+    }
+
+    public MultiChannelScheduler(RDBI rdbi, String redisPrefixKey) {
+        this(rdbi, redisPrefixKey, System::currentTimeMillis);
+    }
+
+    /**
+     * see {@link AbstractDedupJobScheduler#schedule(String, String, int)}
+     *
+     * @return true if the job was scheduled.
+     *         false indicates the job already exists in the ready queue.
+     */
+    public boolean schedule(String channel, String tube, final String job, final int runInMillis) {
```

Comment:
If I'm understanding correctly, there's a tube and that tube has multiple channels within that tube.  If that's right, I'd prefer to to not have to repeat the tube param through all the apis methods in this class.  Instead, how about 
```
MultiChannelScheduler scheduler = MultiChannelScheduler.builder()
                                                                                                   .rdbi()
                                                                                                   .redisPrefixKey()
                                                                                                   .channel()
                                                                                                   .clock()
                                                                                                   .build();

scheduler.schedule(String tube, String job, int runInMillis)

```

That way there's less of a chance that api consumers will inadvertently swap their channel and tube strings.

Reply:
yeah that's a good point, but i'm not sure about the use case from, say omega, which is going to be constantly scheduling jobs for different tubes + channels, is the overhead of maintaining a cache of schedulers per tube (or channel (or both)) worth it?

alternately i could create a type for Channel that's a simple string wrapper.. i'll consider it some more.